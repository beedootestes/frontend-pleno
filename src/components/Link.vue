<template>
  <div class="link" @click="$emit('click')">
    <router-link :to="to">
      <slot></slot>

      {{ title }}
    </router-link>
  </div>
</template>

<script setup lang="ts">
  defineProps({
    title: {
      type: String,
      required: false
    },

    to: {
      type: String,
      required: true
    }
  })
</script>

<style lang="scss" scoped>
  .link {
    color: $color-inactive;

    &.active {
      color: #f25700;
    }

    a {
      display: flex;
      flex-direction: column;
      line-height: 0;
      font-weight: unset;
    }
  }
</style>