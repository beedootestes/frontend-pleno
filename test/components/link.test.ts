import {mount, VueWrapper} from "@vue/test-utils";
import {describe, expect, test} from "vitest";
import Link from "../../src/components/Link.vue";

describe('Test Link.vue', () => {
  test('mount component', async () => {
    expect(Link).toBeTruthy()

    const wrapper: VueWrapper<Link> = mount(Link, {
      props: {
        title: 'exams',
        to: 'exams'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.text()).toContain('exams')
  })
})
