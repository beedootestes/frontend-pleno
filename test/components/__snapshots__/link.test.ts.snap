// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test Link.vue > mount component 1`] = `
"<div data-v-e871e9d7=\\"\\" class=\\"link\\">
  <router-link data-v-e871e9d7=\\"\\" to=\\"exams\\"> exams</router-link>
</div>"
`;
